### PR TITLE
Changes made

### PR DESCRIPTION
**Pull request by harshvaghasiya-webelight**
Shorthand method done
folder structure done
routes are changed
for formatting i using preetieer but i think due to space issue in git hub it not showing in vs code it is formated
port variable edited
name correcting done
extra variable are deleted and direct pass the value

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ app.use(router);
 startServer()
 
 //  Start server after cheaking connnection
-app.listen(process.env.SERVER_PORT || 3000, () => {
+app.listen(process.env.PORT, () => {
```
Comment (harshvaghasiya-webelight) [inline src/app.js:20]: Port name edited

File: src/controller/controller.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const model = require("../models/user");
+
+//  Add new data
+exports.adddata = async (req, res) => {
+  try {
+    await model.create(({ userName, email, birthdate, age, password } = req.body));
```
Comment (harshvaghasiya-webelight) [inline src/controller/controller.js]: shorthand method used and remove extra variable and direct passed the value for both adddata and update data 

await model.create(({ userName, email, birthdate, age, password } = req.body));

File: src/models/user.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+const schema = mongoose.Schema;
+const saltRounds = 10;
+
+const userSchema = new schema({
```
Comment (harshvaghasiya-webelight) [inline src/models/user.js:6]: name correction done

File: src/routes/userRouter.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
+
+const controller = require("../controller/controller");
+const route = express.Router();
+
+// Add New User
```
Comment (harshvaghasiya-webelight) [inline src/routes/userRouter.js:6]: routes are changed as per your suggestion

File: src/controller/controller.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      );
+  } catch (err) {
+    res.send(err.message + "Fetching data is unsucessfull").status(400);
+  }
+};
+
+// Update the data using id
+
+exports.updatedata = async (req, res) => {
+  try {
```
Comment (harshvaghasiya-webelight) [inline src/controller/controller.js]: for the formatting i used the pretieer and it is well formated in vs code i think due to space isssue some lines are showing in git hub

File: package.json [hunk 6]
```diff
@@ -1,11 +1,11 @@
 {
-  "name": "demo",
+  "name": "demo-project",
```
Comment (harshvaghasiya-webelight) [inline package.json:2]: Folder structure done

File: src/controller/controller.js (hunk 2 above)
Comment (Anshshah-1609) [reply src/controller/controller.js]: not like that

Comment (Anshshah-1609) [reply src/controller/controller.js]: `const { name, email, age, birthdate, password } = req.body;`

    await model.create({
      name,
      email,
      age,
      birthdate,
      password,
    });`

File: src/controller/controller.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
+const model = require("../models/user");
```
Comment (Anshshah-1609) [inline src/controller/controller.js]: `model` => `User`
specify name of the model

File: src/controller/controller.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  }
+};
+
+// Update the data using id
+
+exports.updatedata = async (req, res) => {
+  try {
+    await model.findOneAndUpdate(
+      { _id: req.params.userId },
+      ({ userName, email, birthdate, age, password } = req.body)
```
Comment (Anshshah-1609) [inline src/controller/controller.js]: same comment as adddata method

**Review by Anshshah-1609 — CHANGES_REQUESTED** [review]

File: src/models/user.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    min: 0,
+    max: 100,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+});
+// To Encrypt Password while adding new data
+userSchema.pre("save", async function (value) {
```
Comment (Anshshah-1609) [inline src/models/user.js:41]: use arrow function syntax

Comment (harshvaghasiya-webelight) [reply src/models/user.js:41]: Arrow function not work due to this is access globle data so it's not working hear

**Pull request merged**